### ISSUE TITLE
Proxy fails to connect to cluster in KRaft mode when broker ports restricted to 1

### ISSUE THREAD

**Issue by katheris** [kroxylicious/kroxylicious#1045]
**Describe the bug**
When using the Kroxylicious proxy to connect to a Kafka cluster that is running in KRaft mode it is incorrectly requesting ports. Specifically I saw this when running with a single combined node and setting `numberOfBrokerPorts` to 1.

**To Reproduce**
1. Follow the steps in the Kafka quickstart to start a single Kafka node in KRaft mode: https://kafka.apache.org/quickstart
2. Start Kroxylicious with `numberOfBrokerPorts` set to 1
3. Try to list Kafka topics: `./bin/kafka-topics.sh --bootstrap-server localhost:9192 --list`
4. The command hangs and Kroxylicious logs the error "Cannot generate broker address for node id 1"
5. Performing the same steps with a single Zookeeper node and single Kafka node in Zookeeper mode works as expected and does not fail.

**Expected behavior**
The proxy should successfully use the ports 9192 and 9193 to proxy traffic.

**YAML files and logs**

Config yaml used:
```
adminHttp:
  endpoints:
    prometheus: {}
virtualClusters:
  demo:
    targetCluster:
      bootstrap_servers: localhost:9092
    clusterNetworkAddressConfigProvider:
      type: PortPerBrokerClusterNetworkAddressConfigProvider
      config:
        bootstrapAddress: localhost:9192
        numberOfBrokerPorts: 1
    logNetwork: false
    logFrames: false
```

Error logged:
```
2024-02-28 10:00:49 <KQueueEventLoopGroup-5-7> WARN  io.kroxylicious.proxy.internal.FilterHandler:293 - [id: 0xee3cf6a7, L:/127.0.0.1:49790 - R:localhost/127.0.0.1:9092]: Filterresponse for BrokerAddressFilter@376749768 METADATA ended exceptionally - closing connection. Cause message java.lang.IllegalArgumentException: Cannot generate broker address for node id 1 as port 9194 would fall outside port range 9193-9193 that is defined for provider with downstream bootstrap localhost:9192)
```

**Comment by robobario** [comment]
A related issue is #902 which describes the same issue happening if you introduce a broker with id >= numberOfBrokerPorts

The port-per-broker scheme assumes broker ids start at 0 and `numberOfBrokerPorts` >= number of brokers. To deterministically select a port for a broker we do `bootstrapPort + brokerId`.

The workaround is to increase numberOfBrokerPorts, but as in the issue above this may not be reasonable if the setup involves rotating in new broker ids regularly. We'd need to offer a huge number of ports.

One idea put forward to solve this (without adding in any co-ordination between Kroxylicious instances) is to modulo the node id. So with `numberOfBrokerPorts == 8` nodeId 1 and 9 would map to the same port. Assuming the node ids are monotonically increasing and you set `numberOfBrokerPorts` to a high enough number it would probably work in a lot of scenarios.

We could also add potentially add in a startBrokerId.

**Comment by k-wall** [comment]
One effect of the Strimzi's project's move to node pools coupled with https://cwiki.apache.org/confluence/display/KAFKA/KIP-919%3A+Allow+AdminClient+to+Talk+Directly+with+the+KRaft+Controller+Quorum+and+add+Controller+Registration might mean that a Kroxylicous instance has to content with broker ids and controller ids drawn from separate ranges.  That's not going to work well with the current scheme.

This is making me think we need some kind of pluggable port mapping API which allows for different behaviours.

**Comment by robobario** [comment]
> That's not going to work well with the current scheme.

That might break the modulo idea too, if it's more likely we have overlapping ids since the ranges are separate and evolve separately.

**Comment by franvila** [comment]
Triaged: we need some simple fix for that on 0.5.0

**Comment by robobario** [comment]
The simple fix discussed on Triage is to add an `initialBrokerId` or `startBrokerId` property so you can at least override the beginning of the range from 0 to 1, rather than increasing the `numBrokerPorts`
